packaging/selinux/microshift.fc

```diff
@@ -1,6 +1,8 @@
 # vim: sw=8:ts=8:et
-/var/run/secrets/kubernetes.io/serviceaccount(/.*)?				gen_context(system_u:object_r:container_runtime_tmpfs_t,s0)
-/var/lib/microshift/certs/ca-bundle(/.*)?                                       gen_context(system_u:object_r:container_file_t,s0)
-/usr/local/bin/microshift                                                  --	gen_context(system_u:object_r:container_runtime_exec_t,s0)
-/usr/bin/microshift                                                        --	gen_context(system_u:object_r:container_runtime_exec_t,s0)
-/var/hpvolumes(/.*)?            						gen_context(system_u:object_r:container_file_t,s0)
+/var/run/secrets/kubernetes.io/serviceaccount(/.*)?             gen_context(system_u:object_r:container_runtime_tmpfs_t,s0)
+/var/lib/microshift/certs/ca-bundle(/.*)?                       gen_context(system_u:object_r:container_file_t,s0)
+/var/lib/microshift-backups(/.*)?                           	gen_context(system_u:object_r:container_var_lib_t,s0)
```

Comment:
I see the certs paths on line 3 are given different settings. Do we need to worry about that for backups, too, or will the files get the right settings when they're restored to the running location?

Do we need to worry that the backups are reflink copies of the original files? I think if the files are changed the links are broken and new data is written, but does the reflink allow _reading_ data that might not otherwise be readable?

Do we need other files in /var/lib/microshift to have specific contexts?

Reply:
>I see the certs paths on line 3 are given different settings. Do we need to worry about that for backups, too, or will the files get the right settings when they're restored to the running location?

Ah good call out, let me verify the restore for the context, given that we didn't have a transition rule prior I don't know if that desired context was even being applied during restore before this change.

> Do we need to worry that the backups are reflink copies of the original files? I think if the files are changed the links are broken and new data is written, but does the reflink allow reading data that might not otherwise be readable?

Regarding the reflink, given that SELinux operates with just labels I don't know if reflinks change at all. We're just enforcing who can read what files, so here we're saying that containers can't read files in microshift-backups. Right now I don't see any policy denials in our backup and restore test or our failed staged test (which does a restore).

> Do we need other files in /var/lib/microshift to have specific contexts?

I don't even fully think we need to have the granularity we currently do around the certs bundle, that might be a hold out but I don't have the full context on that one just yet. I think we'll need to add granularity as we identify situations where access needs to be granted, but given the current policy doesn't seem to be causing denials, I'm not sure it's needed yet.